src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java

```diff
@@ -121,6 +144,204 @@ public void importRepositories(final ProgrammingExercise templateExercise, final
+
+    private void copyExerciseContentToRepositoryForHaskellExercise(Repository repository, RepositoryType repositoryType, Path exerciseDir) throws IOException {
+        boolean testRepo;
+        if (repositoryType == RepositoryType.TESTS) {
+            repositoryService.createFolderRecursively(repository, "test");
+            testRepo = true;
+        }
+        else {
+            testRepo = false;
+            repositoryService.createFolderRecursively(repository, "src");
+        }
+
+        for (Path file : retrieveRepositoryContentPathsHaskell(exerciseDir, repositoryType)) {
+            if (testRepo) {
+                repositoryService.createFile(repository, "test/" + file.getFileName(), Files.newInputStream(file));
+            }
+            else {
+                repositoryService.createFile(repository, "src/" + file.getFileName(), Files.newInputStream(file));
+            }
+
+        }
+    }
+
+    private void copyExerciseContentToRepositoryForJavaOrKotlinExercise(Repository repository, RepositoryType repositoryType, Path exerciseDir, ProgrammingExercise newExercise,
+            String oldPackageName) throws IOException {
+        boolean testRepo;
+        if (repositoryType == RepositoryType.TESTS) {
+            repositoryService.createFolderRecursively(repository, "test/" + newExercise.getPackageFolderName());
+            testRepo = true;
+        }
+        else {
+            testRepo = false;
+            repositoryService.createFolderRecursively(repository, "src/" + newExercise.getPackageFolderName());
+        }
+        Map<String, String> replacePackageName = Map.of(oldPackageName, newExercise.getPackageName());
+
+        retrieveRepositoryContentPathsJavaKotlin(exerciseDir, repositoryType).forEach(file -> {
+            try {
+                if (testRepo) {
+                    repositoryService.createFile(repository, "test/" + newExercise.getPackageFolderName() + "/" + file.getFileName(), Files.newInputStream(file));
+                }
+                else {
+                    repositoryService.createFile(repository, "src/" + newExercise.getPackageFolderName() + "/" + file.getFileName(), Files.newInputStream(file));
+                }
+                var path = repository.getLocalPath();
+                path = testRepo ? path.resolve("test") : path.resolve("src");
+                var absolutePath = path.toAbsolutePath();
+                fileService.replaceVariablesInFileRecursive(absolutePath.toString(), replacePackageName);
+            }
+            catch (IOException e) {
+                throw new InternalServerErrorException("Could not copy exercise content to repository");
+            }
+
+        });
+    }
+
+    private void deleteSourceAndTestFilesInRepository(Repository repository) {
+        // only the root level dir exists, delete all existing files
+        if (gitService.listFilesAndFolders(repository).entrySet().stream().filter(e -> e.getValue() == FileType.FOLDER).count() == 1) {
```

Comment:
```suggestion
        boolean onlyRootDirectoryExists = gitService.listFilesAndFolders(repository).entrySet().stream().filter(e -> FileType.FOLDER.equals(e.getValue())).count() == 1;
        if (onlyRootDirectoryExists) {
```
easier to understand condition + null-safe equals

Reply:
Why is this special case needed?